lib/box/client.go

```diff
@@ -136,19 +136,10 @@ func (err ExitError) Error() string {
 func dial(socketPath string) (net.Conn, error) {
 	conn, err := net.Dial("unix", socketPath)
 	if err != nil {
-		// try to open socketPath to see whether it's a permissions problem
-		_, openErr := os.Open(socketPath)
-		if openErr != nil {
-			if os.IsPermission(openErr) {
-				// socketPath exists but we don't have permission to access it
-				return nil, checkError(
-					trace.Wrap(err, "Permission denied connecting to planet socket. Try `sudo gravity shell`"))
-			}
-			return nil, checkError(
-				trace.Wrap(err, "Unknown error connecting to planet socket"))
+		if isPermissionDenied(err) {
+			return nil, trace.Wrap(err, "Permission denied accessing planet socket. Try `sudo gravity shell`")
```

Comment:
nit: I'd avoid referring to a specific command in the error message as it might not be the original command. This method simply does not have the context to deduce this - simply offer to run the command (whatever it was) with `sudo`.